services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -188,11 +166,122 @@ void testSaveBulkSubmission() throws IOException {
+        bulkSubmissionService.saveBulkSubmission(bulkSubmission);
+
+
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertNotNull(logList.get(0).getFormattedMessage());
+
+        logger.detachAndStopAllAppenders();
+        // Verify the Mock
+        verify(mockGenericDao).persist(bulkSubmission);
+
+        assertTrue(true, EXPECTED_TO_PASS);
+    }
+
+
+
+   @Test
+    void testSaveBulkSubmissionTargetFound() throws IOException {
+        SdtContext.getContext().setRawInXml(Utilities.getRawXml(PATH_TEST_RESOURCES, TEST_XML_VALID_2));
+        ITargetApplication targetApplication = new TargetApplication();
+        IIndividualRequest individualRequestMock = mock(IIndividualRequest.class);
+
+        List<IIndividualRequest> individualRequests = new ArrayList<>();
+        individualRequests.add(individualRequestMock);
+        Logger logger = (Logger) LoggerFactory.getLogger(BulkSubmissionService.class);
+
+        // Set logging level to debug so that afterCompletion logging is captured
+        logger.setLevel(DEBUG);
+
+        // Create appender and add to logger
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+
+        // Mock the serviceRequest fetch
+        final IServiceRequest serviceRequest = new ServiceRequest();
+        when(mockGenericDao.fetch(IServiceRequest.class, 1)).thenReturn(serviceRequest);
+
+        // Activate Mock Generic Dao
+        final IBulkSubmission bulkSubmission = this.createBulkSubmission();
+        when(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).thenReturn(bulkSubmission.getBulkCustomer());
+
+        bulkSubmission.setIndividualRequests(individualRequests);
+
+        final String key =
+            bulkSubmission.getBulkCustomer().getSdtCustomerId() + bulkSubmission.getCustomerReference();
+
+        // Setup concurrency map as if validator had done it.
+        IInFlightMessage inFlightMessage = new InFlightMessage();
+        inFlightMessage.setCompetingThreads(new HashMap<>());
+        inFlightMessage.getCompetingThreads().put(Thread.currentThread(), Thread.currentThread());
+
+        when(mockConcurrencyMap.get(key)).thenReturn(inFlightMessage);
 
-        assertTrue(true, "Expected to pass");
+        // Put a dummy value into the SdtContext
+        SdtContext.getContext().setServiceRequestId(1L);
+
+        when(mockTargetApplicationDao.getTargetApplicationByCode(anyString())).thenReturn(targetApplication);
+        when(individualRequestMock.getRequestStatus()).thenReturn("Rejected");
+
+        when(individualRequestMock.isEnqueueable()).thenReturn(false);
+        // Call the bulk submission service
+        bulkSubmissionService.saveBulkSubmission(bulkSubmission);
+
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertTrue(verifyLog(logList,"Target Application found {}"));
+
+        logger.detachAndStopAllAppenders();
```

Comment:
This method looks like it needs some mock verifications at the end of it to confirm that things have or haven't happened.  For example, if the requests aren't enqueue-able then check that the mock messaging utility enqueueRequest() method hasn't been called (something like ```verify(mockMessagingUtility, never()).enqueueRequest()```).

Reply:
done

Reply:
Unable to verify as this is a privte method.

Reply:
I might be looking at the wrong thing, but the MessagingUtility enqueueRequest() method is public - see https://github.com/hmcts/civil-sdt/blob/master/services/src/main/java/uk/gov/moj/sdt/services/utils/MessagingUtility.java#L76.  You can use never() to check that something has never been called - see https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#7

Reply:
Add another verification to testSaveBulkSubmissionTargetFoundIsEnqueueableSetToFalse() test to check that individualRequestMock.isEnqueueable() was called.

Reply:
done